packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -214,9 +213,13 @@ export class SpinButton extends React.Component<ISpinButtonProps, ISpinButtonSta
                 role="spinbutton"
                 aria-labelledby={label && this._labelId}
                 aria-valuenow={
-                  !isNaN(Number(ariaValueNow)) ? ariaValueNow : !isNaN(Number(value)) ? Number(value) : undefined
+                  typeof ariaValueNow === 'number'
+                    ? ariaValueNow
+                    : value && !isNaN(Number(value)) // Number('') is 0 which may not be desirable
+                    ? Number(value)
+                    : undefined
                 }
-                aria-valuetext={ariaValueText ? ariaValueText : isNaN(Number(value)) ? value : undefined}
+                aria-valuetext={ariaValueText || (!value || isNaN(Number(value)) ? value : undefined)}
```

Comment:
The parentheses actually have to be where they are now, or the logic is incorrect. It won't end up with boolean values because `||` has higher precedence than `?`, so it turns out like this (extra parens added for illustration: `ariaValueText || ( (!value || isNaN(Number(value)) ) ? value : undefined)`.

I may actually change it back to a ternary (despite the first condition being redundant) to reduce confusion.

Reply:
Switched back to a ternary and changed the condition slightly, so the grouping should be clearer now.